app/components/articleSearch/reducer.tsx

```diff
@@ -203,39 +198,52 @@ export function reducer(
         }
       });
 
-      return {
-        ...state,
-        searchItemsToShow: newSearchItemsToShow,
-      };
+      return { ...state, searchItemsToShow: newSearchItemsToShow };
     }
 
-    case ACTION_TYPES.ARTICLE_SEARCH_START_TO_PUT_MY_FILTERS: {
-      return {
-        ...state,
-        isFilterSaveBoxLoading: true,
-      };
+    case ACTION_TYPES.ARTICLE_SEARCH_START_TO_PUT_CURRENT_USER_FILTERS: {
+      return { ...state, isFilterSaveBoxLoading: true };
     }
 
-    case ACTION_TYPES.ARTICLE_SEARCH_FAILED_TO_PUT_MY_FILTERS: {
-      return {
-        ...state,
-        isFilterSaveBoxLoading: false,
-      };
+    case ACTION_TYPES.ARTICLE_SEARCH_FAILED_TO_PUT_CURRENT_USER_FILTERS: {
+      return { ...state, isFilterSaveBoxLoading: false };
     }
 
-    case ACTION_TYPES.ARTICLE_SEARCH_SUCCEEDED_TO_GET_MY_FILTERS: {
-      const payload: Filter[] = action.payload;
+    case ACTION_TYPES.ARTICLE_SEARCH_SUCCEEDED_TO_GET_LOCAL_STORAGE_FILTERS: {
+      const payload: RawFilter[] = action.payload;
+      const migrationFilters = ResponseFilterObjectGenerator(payload);
+
+      return { ...state, myFilters: migrationFilters };
+    }
+
+    case ACTION_TYPES.ARTICLE_SEARCH_SUCCEEDED_TO_GET_CURRENT_USER_FILTERS: {
```

Comment:
now i feel all of this logic is too verbose to understand and use.
i believe we need to refactor all of them. for now, i'll skip review about it.